src/Couchbase.Lite.Shared/API/Document/MutableDictionaryObject.cs

```diff
@@ -219,6 +234,24 @@ public IMutableDictionary SetDictionary(string key, DictionaryObject value)
             return this;
         }
 
+        /// <inheritdoc />
+        public IMutableDictionary SetJSON([NotNull] string json)
+        {
+            JObject jobj = null;
+            try {
+                jobj = JObject.Parse(json);
+            } catch {
+                throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, CouchbaseLiteErrorMessage.InvalidJSON);
+            }
+
+            if (jobj.GetType() == typeof(JArray)) {
+                throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, CouchbaseLiteErrorMessage.InvalidJSON);
+            }
+
+            var blobDict = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
```

Comment:
This can throw if I recall correctly so I think it should be wrapped in try / catch that will throw the same exception as above.

Reply:
done.